src/importexport/musicxml/internal/musicxml/musicxmlsupport.h

```diff
@@ -221,6 +221,20 @@ struct MusicXMLInstrument {
      */
 };
 
+struct InferredPercInstr {
+    int pitch;
+    staff_idx_t staff;
+    String name;
+
+    InferredPercInstr(int pitch, staff_idx_t staff, String name)
+        : pitch(pitch), staff(staff), name(name) {}
+
+    InferredPercInstr()
+        : pitch(-1), staff(muse::nidx), name(u"") {}
+};
+
+typedef std::multimap<Fraction, InferredPercInstr> InferredPercMap;
```

Comment:
This looks a bit weird. The location of an InferredPercInstr is univoquely defined by a tick and a staffIdx (as I'm assuming from the way that you are querying this map above in the function inferredPercInstr).  So it's a bit strange to use the tick as key of a multimap, but the staffIdx as an internal member of the struct, and it makes it confusing to read. You could do one of two things: a) define something like 
```
InferredPercInstrLocation {
   Fraction tick;
   staff_idx_type;
}
```
That way the map doesn't need to be a multimap anymore, but simply std::map, or better `std::unordered_map<InferredPercInstrLocation, InferredPercInstrument>`, which also simplifies the function inferredPercInstr, which can still take the same inputs and return-if-exist `m_inferredPerc(InferredPercInstLocation{tick, staffIdx});`

b) You also put the tick as an inner member, and store them as a simple vector, then the query function will look more or less as it does now, where you scan the elements and compare.

Reply:
Yeah, on a second look there aren't many items in the map ever (6 is the most I've seen), so I'll go with a vector.  Thanks!

Reply:
I've also changed staff index to track index, as there can be one per voice.